cmd/promtool/metrics.go

```diff
@@ -101,28 +92,16 @@ func PushMetrics(url *url.URL, roundTripper http.RoundTripper, headers map[strin
 	return successExitCode
 }
 
-// TODO(bwplotka): Add PRW 2.0 support.
-func parseAndPushMetrics(client *remote.Client, data []byte, labels map[string]string) bool {
+func parseAndPushMetrics(writeAPI *remoteapi.API, data []byte, labels map[string]string) bool {
 	metricsData, err := fmtutil.MetricTextToWriteRequest(bytes.NewReader(data), labels)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, "  FAILED:", err)
 		return false
 	}
 
-	raw, err := metricsData.Marshal()
-	if err != nil {
-		fmt.Fprintln(os.Stderr, "  FAILED:", err)
-		return false
-	}
-
-	// Encode the request body into snappy encoding.
-	compressed, err := compression.Encode(compression.Snappy, raw, nil)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, "  FAILED:", err)
-		return false
-	}
-
-	_, err = client.Store(context.Background(), compressed, 0)
+	// Use remoteapi.Write which handles marshaling and compression internally.
+	// Default to V1 for compatibility.
```

Comment:
This is not really true. This code selects v1, it's not like there is a default or a choice for promtool now.

So let's either:
* Remove this comment and add v2 support in separate PR (e.g. by adding flag what to use)
* Do the flag now in this PR